src/js/view/captionsrenderer.js

```diff
@@ -77,24 +73,53 @@ define([
                     fontSize: Math.floor(size*2)/2 + 'px'
                 });
             }
-            renderCues(true);
+            this.renderCues(true);
         };
 
-        this.renderCues = renderCues;
-
-        function renderCues(updateBoxPosition) {
+        this.renderCues = function(updateBoxPosition) {
             updateBoxPosition = !!updateBoxPosition;
             if(_VTTRenderer) {
                 _VTTRenderer.WebVTT.processCues(window, _currentCues, _display, updateBoxPosition);
             }
-        }
+        };
 
-        function _timeChange(e) {
-            _timeEvent = e;
-            _select(_captionsTrack, _timeEvent);
-        }
+        this.selectCues = function(track, timeEvent) {
+            var cues, pos;
+
+            if (!track || !track.data || !timeEvent) {
+                return;
+            }
+
+            pos = this.getAlignmentPosition(track, timeEvent);
+            if (pos === false) {
+                return;
+            }
+
+            cues = this.getCurrentCues(track.data, pos);
+
+            this.updateCurrentCues(cues);
+            this.renderCues();
+        };
+
+        this.getCurrentCues = function(allCues, pos) {
+            return _.filter(allCues, function (cue) {
+                return pos >= (cue.startTime) && (!cue.endTime || pos <= cue.endTime);
+            });
+        };
+
+        this.updateCurrentCues = function(cues) {
+            // Render with vtt.js if there are cues, clear if there are none
+            if (!cues.length) {
+                _currentCues = [];
+            } else if (_.difference(cues, _currentCues).length) {
+                // _captionsWindow.className = 'jw-captions-window jw-reset jw-captions-window-active';
```

Comment:
Remove this comment and cleanup dead code (references to jw-captions-window-active in less). If display is none in . jw-captions-window how is the window displayed?